logs_api/lib/opentelemetry/logs/logger_provider.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Logs
+    # No-op implementation of a logger provider
+    class LoggerProvider
+      # Returns a {Logger} instance.
+      #
+      # @param name [optional String] Instrumentation package name
+      # @param version [optional String] Instrumentation package version
+      # @param schema_url [optional String] Schema URL recorded in the emitted telemetry.
+      # @param attributes [optional Hash] Specifies the instrumentation scope
+      #  attributes to associate with emitted telemetry.
+      #
+      # @return [OpenTelemetry::Logs::Logger]
+      #
+      # @api public
+      def logger(name: nil, version: nil, schema_url: nil, attributes: {})
+        @logger ||= OpenTelemetry::Logs::Logger.new
+      end
```

Comment:
I was chatting with @fallwith about this method. He asked something I hope folks more versed in the spec can help us answer:

> This method takes 4 named arguments that are all unused. Should the API version of this method not have them, have them with leading underscore-based names to signify non-use, or is the current state intentional for copying and pasting for non-no-op implementations?

Reply:
Not a Ruby person, so I'm not really one to say what would seem right to a Ruby developer.. but I did note that Ruby's trace API does not use leading underscores:

https://github.com/open-telemetry/opentelemetry-ruby/blob/ccad50f9e5f04ba61fa991cabe795853ec87060b/api/lib/opentelemetry/trace/tracer_provider.rb#L17-L19

Also, this probably makes sense because this API will be consumed by instrumentation authors and whether or not the arguments will be used is unknown to them. Their use depends on an application author configuring the SDK for their application.

Reply:
Also of note.. the trace API does not currently support `schema_url` or `attributes` for tracer provider. Might make sense to add this support across all signals at the same time. Does it make sense to remove these arguments for now from the logger provider?

Reply:
> Also of note.. the trace API does not currently support schema_url or attributes for tracer provider. Might make sense to add this support across all signals at the same time. Does it make sense to remove these arguments for now from the logger provider?

Thanks for catching that! I think it'd make sense to drop them for now, but I'd like to hear more from the OTel Ruby team. 

cc @mwear, @robbkidd, @robertlaurin

Reply:
I haven't had time to look at the PR as a whole, but I can answer this question with my thoughts (which may differ from others). I think we should stay consistent in our `*Provider` APIs. Given that we've gone with positional arguments for tracing, we should probably do the same for logs. I would also say that we should omit schema url for now, and add it to all signals, when the time is right.

Reply:
Thanks, @mwear! I've changed the code to be more consistent with the`TracerProvider` API, pulling out `schema_url` and `attributes` in dc966f0a153a386f91614742c3c6264422b9e6ca